docs/deploying.rst

```diff
@@ -0,0 +1,328 @@
+Deploying a workflow
+====================
+
+If a Toil workflow is run on a single machine (that is, single machine mode),
+there is nothing special you need to do. You change into the directory
+containing your user script and invoke it like any Python script:
+
+.. code-block:: console
+
+   $ cd my_project
+   $ ls
+   userScript.py …
+   $ ./userScript.py …
+
+This assumes that your script has the executable permission bit set and
+contains a *shebang*, i.e. a line of the form
+
+::
+
+   #!/usr/bin/env python
+
+Alternatively, the shebang can be omitted and the script invoked as a module
+via
+
+::
+
+   $ python -m userScript
+
+in which case the executable permission is not required either. Both are common
+methods for invoking Python scripts.
+
+The script can have dependencies, as long as those are installed on the machine,
+either globally, in a user-specific location or in a virtualenv. In the latter
+case, the virtualenv must of course be active when you run the user script.
+
+If, however, you want to run your workflow in a distributed environment, on
+multiple worker machines, either in the cloud or on a bare-metal cluster, your
+script needs to be made available to those other machines. If your script
+imports other modules, those modules also need to be made available on the
+workers. Toil can automatically do that for you, with a little help on your
+part. We call this feature *hot-deployment* of a workflow.
+
+Let's first examine various scenarios of hot-deploying a workflow and then take
+a look at :ref:`deploying Toil <deploying_toil>`, which, as we'll see shortly
+cannot be hot-deployed. Lastly we'll deal with the issue of declaring
+:ref:`Toil as a dependency <depending_on_toil>` of a workflow that is packaged
+as a setuptools distribution.
+
+
+Hot-deployment without dependencies
+-----------------------------------
+
+If your script has no additional dependencies, i.e. imports only modules that
+are shipped with Python or Toil, only your script needs to be hot-deployed.
+Both Python and Toil are assumed to be present on all workers. Toil takes your
+script, stores it in the job store and just before the jobs in your script are
+about to be run on a worker machine, your script will be saved to a temporary
+directory on the worker and loaded into the Python interpreter from there.
+
+In this scenario, the script is invoked as follows:
```

Comment:
These invocations are missing the activation of the virtualenv. Let's add an example of creating a virtualenv (using `--system-site-packages` so mesos can be discovered) and activation of the virtualenv before running the script.